src/types/profile.js

```diff
@@ -942,15 +942,34 @@ export type Profile = {|
 |};
 
 type SerializableThread = {|
-  ...$Diff<Thread, { stringTable: UniqueStringArray }>,
+  ...$Diff<Thread, { stringTable: UniqueStringArray, samples: SamplesTable }>,
   stringArray: string[],
+  samples: SerializableSamplesTable,
+|};
+
+export type SerializableSamplesTable = {|
+  ...$Diff<SamplesTable, { time: Milliseconds[] }>,
+  time?: Milliseconds[],
+  timeDeltas: Milliseconds[],
+|};
+
+export type SerializableCounterSamplesTable = {|
+  ...$Diff<CounterSamplesTable, { time: Milliseconds[] }>,
+  time?: Milliseconds[],
+  timeDeltas: Milliseconds[],
```

Comment:
Hmm, if you put an optional `time` field here, then I think it would make sense for the `timeDeltas` field to also be optional.